### DISCUSSION TITLE
Spatie media library plugin: Use multiple fields in one form

### DISCUSSION THREAD

**Discussion by dmitrijsmihailovs** [filamentphp/filament#10921] [Help]
I am trying to use multiple file fields, but when I upload file in one field, create/save form, it shows up in all the fields.
Maybe I am missing some configuration or is this not possible?

```
SpatieMediaLibraryFileUpload::make('logo')
                    ->image()
                    ->collection('channels'),
                
 SpatieMediaLibraryFileUpload::make('icon')
                    ->image()
                    ->collection('channels'),

SpatieMediaLibraryFileUpload::make('background')
                    ->image()
                    ->collection('channels'),
```

![Screenshot Spatie Media Library](https://github.com/filamentphp/filament/assets/6555569/92e8af11-8b87-4d4b-af56-ee56a8d75a37)

**Comment by leandrocfe** [comment]
Maybe because they are in the same collection

**Reply by dmitrijsmihailovs** [reply]
Looks like that was it, though it is kinda weird, that it is not saved by the field name.

**Reply by danharrin** [reply]
but Spatie doesnt save the files separately, it saves them into the same collection, that's what collections are used for
